index.js

```diff
@@ -3,29 +3,67 @@ import pMemoize from 'p-memoize';
 import delay from 'delay';
 
 const getDevice = pMemoize(async () => {
-	if (process.platform !== 'darwin') {
-		throw new Error('macOS only');
+	if (process.platform == 'darwin') {
+		const { stdout } = await execa('networksetup', ['-listallhardwareports']);
+		const result = /Hardware Port: Wi-Fi\nDevice: (?<device>en\d)/.exec(stdout);
+		if (!result) {
+			throw new Error('Couldn\'t find a Wi-Fi device');
+		}
+
+		return result.groups.device;
+	} else if (process.platform == 'win32') {
+		const { stdout } = await execa('powershell', [
+			'-command',
+			'Get-NetAdapter -Name Wi-Fi | Format-List -Property PnPDeviceID',
+		]);
+
+		const result = /(?<=PnPDeviceID : ).*/.exec(stdout);
+
+		if (!result) {
+			throw new Error('Couldn\'t find a Wi-Fi device');
+		}
+
+		return result[0];
+	} else {
+		throw new Error('macOS or Windows only');
 	}
-
-	const {stdout} = await execa('networksetup', ['-listallhardwareports']);
-
-	const result = /Hardware Port: Wi-Fi\nDevice: (?<device>en\d)/.exec(stdout);
-	if (!result) {
-		throw new Error('Couldn\'t find a Wi-Fi device');
-	}
-
-	return result.groups.device;
 });
 
-const isOn = async device => {
-	const {stdout} = await execa('ifconfig', [device]);
-	return stdout.includes('status: active');
+const isOn = async (device) => {
+	if (process.platform == 'darwin') {
+		const { stdout } = await execa('ifconfig', [device]);
+		return stdout.includes('status: active');
+	} else if (process.platform == 'win32') {
+		const { stdout } = await execa('powershell', [
+			'-command',
+			'Get-NetAdapterAdvancedProperty -Name Wi-Fi -RegistryKeyword RFOff -AllProperties | Format-List -Property RegistryValue',
+		]);
+		const status = /(?<=\{).+?(?=\})/.exec(stdout);
+		return !parseInt(status);
+	}
 };
 
 const toggleDevice = async (device, turnOn) => {
-	await execa('networksetup', ['-setairportpower', device, (turnOn ? 'on' : 'off')]);
-
-	await delay(100);
+	if (process.platform == 'darwin') {
+		await execa('networksetup', [
+			'-setairportpower',
+			device,
+			turnOn ? 'on' : 'off',
+		]);
+		await delay(100);
+	} else if (process.platform == 'win32') {
+		var setStatus = turnOn ? 0 : 1;
+		//console.log("Testing..."); // Odd Issue: Without this log statement, tests fail on Windows 10
+
+		await execa(
+			'powershell -command Start-Process PowerShell -Verb RunAs -WindowStyle Hidden',
+			[
+				`-ArgumentList "Set-NetAdapterAdvancedProperty -Name Wi-Fi -RegistryKeyword RFOff -AllProperties -RegistryValue ${setStatus}"`,
+			]
+		);
+		await delay(6000);
```

Comment:
This is very arbitrary. Would be better to poll for the status.

Reply:
I've figured out that the PowerShell won't release the command until it finishes. So, I kept the original logic of `await delay(100)`